system/service/apigw_filter.go

```diff
@@ -18,15 +17,17 @@ type (
 		store     store.Storer
 		ac        routeAccessController
 		route     *apigwRoute
+		ags       apiGwService
 	}
 )
 
-func Filter() *apigwFilter {
+func Filter(ags apiGwService) *apigwFilter {
```

Comment:
Why use the service as a new fields, since the apigw service is a singleton and can be used as before (`.Service()`)?

Reply:
Yeah, I agree I can revert the changes.
The reason which leads me to this is when I had to use service multiple times in same method then either I will have to create a local instance of service in the method or make service available to the struct but we also have 3 `apigw_*` structs hence I created an interface for apigw service and aligned it with rest of the implementations.